File: CloudModernisation/Models/CMSData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CloudModernisation.Models
{
    public class CMSData
    {
        public ApplicationDetails ApplicationDetails { get; set; }
        public SourceCodeDetails SourceCodeDetails { get; set; }
        public HostingDetails HostingDetails { get; set; }
        public DeploymentDetails DeploymentDetails { get; set; }
        public ServiceDetails ServiceDetails { get; set; }
        public IngressDetails IngressDetails { get; set; }
        public JenkinsDetails JenkinsDetails { get; set; }
    }
}

File: src/app/modals/CMSData.ts
import { AppDetails } from './appDetails';
import { SourceCodeDetails } from './sourceCodeDetails';
import { HostingDetails } from './hostingDetails';
import { DeploymentDetailsAKS } from './deploymentDetailsAKS';
import { ServiceDetailsAKS } from './serviceDetailsAKS';
import { IngressDetails } from './ingressAKSDetails';
import { JenkinServerDetails } from './jenkinsServerDetails';


export class CMSData {
  ApplicationDetails: AppDetails;
  SourceCodeDetails: SourceCodeDetails;
  HostingDetails: HostingDetails;
  DeploymentDetails: DeploymentDetailsAKS;
  ServiceDetails: ServiceDetailsAKS;
  IngressDetails: IngressDetails;
  JenkinsDetails: JenkinServerDetails;
  SessionID: string;
  PartitonKey: string;
}
